pkg/parser/violations.go

```diff
@@ -48,6 +48,11 @@ func generateFileViolations(file *os.File, rules []*rule.Rule) (*result.FileResu
 		}
 
 		for _, r := range rules {
+			// Check the filename itself for violations
+			if line == 1 {
```

Comment:
I like the idea, but I think it might cause some confusion with the way this is surfaces as a violation. It might make it seem like a false-positive on line 1 if line 1 doesn't actually have a violation but the filename does.

Reply:
That's true! I can go take a look at how and if reviewdog can handle file-level comments/tags. Do you know?

Reply:
I'm not sure how reviewdog would handle that, or even github actions for that matter. I think just using a different description for the violation might be good enough. My PR https://github.com/get-woke/woke/pull/10 implements a different description. So file path violation descriptions would be prefixed with `Filename violation:`. Hopefully that will make it clear. I'm also thinking of making this feature opt-in.